android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -697,13 +726,30 @@ public void pickLabel(float posX, float posY) {
     }
 
     /**
+     * Query the map for a {@link Marker} at the given screen coordinates; results will be returned
+     * in a callback to the object set by {@link #setMarkerPickListener(MarkerPickListener)}
+     * @param posX The horizontal screen coordinate
+     * @param posY The vertical screen coordinate
+     */
+    public void pickMarker(float posX, float posY) {
+        if (markerPickListener != null) {
+            checkPointer(mapPointer);
+            nativePickMarker(mapPointer, posX, posY, markerPickListener);
+        }
+    }
+
+    /**
      * Adds a {@link Marker} to the map which can be used to dynamically add points and polylines
      * to the map.
      * @return Newly created {@link Marker} object.
      */
     public Marker addMarker() {
         checkPointer(mapPointer);
         long markerId = nativeMarkerAdd(mapPointer);
+
+        Marker marker = new Marker(mapView.getContext(), markerId, this);
```

Comment:
Should this `Marker` instance now be the return value below instead of returning a separate instance?